### ISSUE TITLE
Homeseer HS-WS100+

### ISSUE THREAD

**Issue by gotwalt** [tonesto7/homebridge-smartthings#168]
Added [a new Homeseer HS-WS100+](https://shop.homeseer.com/collections/lighting/products/homeseer-hs-ws100-z-wave-plus-scene-capable-wall-switch?variant=20302954182) z-wave switch to my Smartthings installation using the [DarwinsDen device handler](https://github.com/DarwinsDen/SmartThingsPublic/blob/master/devicetypes/darwinsden/ws100-switch.src/ws100-switch.groovy). Works great - perhaps the best switch I've tried so far.

Does not, however, work with Homebridge. Log dump below. Anything else I can provide or work on to correct this?

`39m Device Skipped - Group button, Name Bedroom Chandelier, ID c4744a11-2c5a-4de4-9716-21f7d647cb72, JSON: {"name":"Bedroom Chandelier","basename":"Z-Wave Switch","deviceid":"c4744a11-2c5a-4de4-9716-21f7d647cb72","status":"ONLINE","manufacturerName":null,"modelName":null,"lastTime":"2017-03-25T22:05:46+0000","capabilities":{"Switch":1,"Polling":1,"Configuration":1,"Refresh":1,"Indicator":1,"Button":1,"Sensor":1,"Actuator":1},"commands":{"on":[],"off":[],"poll":[],"configure":[],"refresh":[],"indicatorWhenOn":[],"indicatorWhenOff":[],"indicatorNever":[],"tapUp2":[],"tapDown2":[],"tapUp3":[],"tapDown3":[],"holdUp":[],"holdDown":[]},"attributes":{"switch":"off","indicatorStatus":"when on","button":"pushed","numberOfButtons":null}}`

**Comment by brendanjerwin** [comment]
I'm interested in this as well. I've got 10 of these switches in a fresh installation and would love to be able to start leveraging their extra capabilities to trigger scenes in HomeKit.

**Comment by pdlove** [comment]
Anyone have one they'd like to send me? I'm having a hard time with why this doesn't currently map to a switch and, more importantly, I don't see the ability to map the tap/hold features to different attributes for passing to HomeKit more than just on/off.

**Comment by brendanjerwin** [comment]
Sure. I'll loan you one of mine. I'll even cover shipping to and from. brendanjerwin@gmail.com

**Comment by mlehrian** [comment]
Any update on this?  I just installed about 12 of these.

**Comment by brendanjerwin** [comment]
Well. I flaked out and didn’t send one for testing... my last one uninstalled turned out to be DOA. :(

**Comment by chebalski** [comment]
I'm also having the same device skipped issue for the WD100+. Funny thing is, I have a mix of these switches as well as the dimmers (WS100+) and the dimmers are added to homebridge just fine. Here's another example if it helps:

[2018-3-26 16:43:10] [SmartThings] Device Skipped - Group  button, Name Kitchen Counter, ID 263d73ef-06b7-497d-b16f-9f75bdbf4070, JSON: {"name":"Kitchen Counter","basename":"WS100+ Switch","deviceid":"263d73ef-06b7-497d-b16f-9f75bdbf4070","status":"ACTIVE","manufacturerName":null,"modelName":null,"lastTime":"2018-03-26T23:42:21+0000","capabilities":{"Switch":1,"Polling":1,"Configuration":1,"Refresh":1,"Indicator":1,"Button":1,"Sensor":1,"Actuator":1},"commands":{"on":[],"off":[],"poll":[],"configure":[],"refresh":[],"indicatorWhenOn":[],"indicatorWhenOff":[],"indicatorNever":[],"tapUp2":[],"tapDown2":[],"tapUp3":[],"tapDown3":[],"holdUp":[],"holdDown":[]},"attributes":{"switch":"off","indicatorStatus":"when off","button":"pushed","numberOfButtons":8}}

**Comment by kosmicken** [comment]
Any update on this? I have one that won't show in Homebridge at all, but all my GE switches work fine. I don't need HomeKit to see the double and triple taps since SmarThings will handle that. But right now I can't tell Siri to turn on and off that one light, and it's super annoying.

**Comment by mlehrian** [comment]
As a workaround, I created virtual switches in SmartThings and mapped them to the HomeSeer switch actions.  The virtual switches show up via Homebridge so you can control them via Siri.  It's a little more work and you don't get status updates back in HomeKit, but they are Siri enabled.

**Comment by kosmicken** [comment]
That's not a bad idea!  I never thought about virtual switches because I never really "got" what they are used for.  This makes perfect sense.

I created a virtual switch in my ST app.  Now for a dumb question: How do I configure what actions are triggered when the switch is turned on and off?  I can't figure it out and a quick google search so far hasn't given me the answer. (I'm sure it's out there if I keep looking.)

**Comment by mlehrian** [comment]
I also used the Smart Lights SmartApp to configure what each virtual switch should do.

It turns out this approach has a couple of unexpected advantages for me.  First, I can easily group sets of lights together.  My kitchen has 6 different light switches, but now I can just say, "Hey Siri, turn off the kitchen lights."  Second, I can control the vocabulary of the switches that are exposed to Siri.  With so many switches it gets very complicated to precisely name each switch, and then to remember them all for access by Siri.  Virtual switches let me trim the set of devices I want to expose to Siri and adjust the naming accordingly.

So, now I have a virtual switch called lights for every room, and it doesn't matter how many lighting devices are in the room or what they're called, I can always just say, "Hey Siri, turn off the lights in the library." or whatever the room name happens to be.  I never have to think about whether the room has a lamp or a chandelier or recessed lights.

**Comment by kosmicken** [comment]
I definitely see the advantage of being able to control multiple devices with a single voice command or tap of a button. If that’s all you’re looking to do (for devices other than the HS switches), I believe the Home app supports this natively. You can group accessories and then all the devices that are grouped together will be controlled with one button or by telling Siri the name of that button. Additionally, I am pretty sure (I will have to test when I get home) that even when devices are not grouped in this manner, if they are all assigned to a room, you can just tell Siri to turn on or off the name of the room and all devices in the room will respond.

**Comment by kosmicken** [comment]
So, I have some very cool news to report after setting this up and testing it.  I created a second rule in the Smart Lighting app that does the opposite of of the first: It turns on the virtual switch when the real one is turned on (and off when it's turned off).  With this rule in place, the virtual switch in the Apple Home app DOES report actual status of the physical HS switch, almost instantly.

I tested controlling all lights in a room with Siri and it works exactly as I thought.  Saying, "Hey Siri, turn on [room name] lights" turns on all lights in that room, whether they are grouped together in the Home app or night. (Note, grouping devices together and assigning devices to a room are not the same thing. Grouping allows multiple devices to be controlled by one button.)  This voice command only controls devices that are configured as a light. If you have fans or switches in addition to lights and want to turn on or off EVERYTHING in the room just say, "Hey Siri, turn on [room name]."
